applications/tari_console_wallet/src/ui/state/wallet_event_monitor.rs

```
@@ -77,6 +80,26 @@ impl WalletEventMonitor {
                             Err(e) => error!(target: LOG_TARGET, "Error reading from Transaction Service event broadcast channel"),
                         }
                     },
+                    result = connectivity_events.select_next_some() => {
+                        match result {
+                            Ok(msg) => {
+                                trace!(target: LOG_TARGET, "Wallet Event Monitor received wallet event {:?}", msg);
+                                     match (*msg).clone() {
+                                        ConnectivityEvent::PeerDisconnected(_) |
+                                        ConnectivityEvent::ManagedPeerDisconnected(_) |
+                                        ConnectivityEvent::PeerConnected(_) |
+                                        ConnectivityEvent::PeerBanned(_) |
+                                        ConnectivityEvent::PeerOffline(_) |
+                                        ConnectivityEvent::PeerConnectionWillClose(_, _) => {
+                                            self.trigger_peer_state_refresh().await;
+                                        },
+                                        /// Only the above variants trigger state refresh
+                                        _ => (),
+                                     }
+                            },
+                            Err(e) => error!(target: LOG_TARGET, "Error reading from Connectivity event broadcast channel"),
```

Unused var `e`? (Is clippy even working?)

nit:
An error! saying "error reading" is not an ideal way to handle this case - the error from `Stream::next` on a tokio broadcast channel happens if 1) There are no senders (disconnected)(This cannot happen because that would be the end of the stream and this block would not execute), or 2) this loop could not read off the channel fast enough and some messages were dropped (lagged).
In this case, ignoring the error is the best option unless you can do something in response to missing some messages.

---

So the only thing you can do is log that there is lagging happening - which could of course help make a slow select loop more noticeable in the logs.